packages/core-data/src/entity-provider.js

```diff
@@ -71,6 +72,23 @@ export default function EntityProvider( { kind, type: name, id, children } ) {
 	return <Provider value={ id }>{ children }</Provider>;
 }
 
+/**
+ * Hook that returns the ID for the nearest
+ * provided entity of the specified type.
+ *
+ * @deprecated since 12.9. Callers should use the useEntityProviderId hook instead.
+ *
+ * @param {string} kind The entity kind.
+ * @param {string} type The entity type.
+ */
+export function useEntityId( kind, type ) {
+	deprecated( 'the @wordpress/core-data useEntityId() hook', {
+		since: '6.0',
+		alternative: 'the @wordpress/core-data useEntityProviderId() hook',
+	} );
+	return useEntityProviderId( kind, type );
```

Comment:
I'm curious why we decided to change the name here. For me "Provider" is just an implementation detail (React context) and what we really do with this hook is to retrieve the "entityId". so I think the old naming is better here.

Reply:
I don't have the context. Adam or Greg might know but they are unavailable.

Would you prefer we just renamed back to `useEntityId()`? That would avoid the whole deprecation thing.

Reply:
For now we can move on with the deprecation notice as the other PR has been through longer discussions and we may miss some context.

Reply:
I disagree, I think we should just restore the old name here.

Reply:
I didn't find any thing clarifying the reasoning for this particular hook name change and since it's a breaking change, I'd rather revert instead of adding a deprecation and then later a deprecation that deprecate the deprecation :)

Reply:
It does seem to be the only instance where "provider" is included. I can't see this being a huge problem.

Probably safest to rename back and then ask Adam, Dennis and Greg to re-review when they are available. If they _really_ want to rename it they can always submit a followup with the appropriate rationale for the next release.

Reply:
OK [here's the alternative](https://github.com/WordPress/gutenberg/pull/39683). Please let me know which route we'd like to take.

Reply:
@youknowriad @getdave @draganescu it was a part of https://github.com/WordPress/gutenberg/pull/39349. Why? The name `useEntityId()` is slightly off semantically. An `Entity` is a higher-level idea that refers to "Comments." "Posts," and such. An entity is not a data container, though, and does not store a numerical ID. That would be an `Entity Record`. In this case, though, the ID does not even come from the record but from the Entity Provider – hence the name change to clarify that.

I missed the fact that this function was a part of the public API, though. I specifically wanted to avoid any public-facing changes and even [made a note about that](https://github.com/WordPress/gutenberg/pull/39349/files#r823750303). Thank you for the prompt reaction and a fix here!

Reply:
I agree that "entity" is not the right semantic here but it's a bit better than "provider" for me. Ideally yet it should be `useEntityRecordId` the provider should be `EntityRecordProvider`. We can make these changes (and deprecate the old ones if deemed necessary)